add instance config to the self-registration page (#102)
neo4j/docs-ops-manager: modules/ROOT/pages/addition/agent-installation/agent-self-signed-certificates.adoc

= Self-signed certificates for agent

[IMPORTANT]
====

> Self-signed certificates are not recommended to be used in production environments.

Reviewer: ```suggestion
It is not recommended to use self-signed certificates in production environments.
```

                              server, eg. --ip=127.0.0.1,192.168.100.55
* -n, --name=<name>         common name to use in generated certificate,
                              eg. --name=localhost
* -o, --output=<output>     target directory, eg. --output=.
  -h, --help                help
----

If you are generating a certificate for an agent on `localhost`, `localhost` is the primary name set as Subject Name field on the certificate and is also used as the name of generated files.
Assume there are also a number of virtual machines that access the server through IP addresses `192.168.10.1` and `172.16.10.1`.
Furthermore, a local DNS alias `nom.example.com` is set up for `localhost`.

In order to generate a self-signed certificate for the above example, execute the following command;

[source, terminal, role=noheader]
----
agent ssc -n localhost \
	-o ./certificates \
	-d nom.example.com \
	-i 192.168.10.1,172.16.10.1
----

It generates a key pair and a self-signed certificate and creates `localhost.cert.pem` and `localhost.key.pem` files inside `./certificates` directory.

You can then use these two files to configure the agents for mTLS authentication with server.

> Similarly, if you need to generate agent specific keys for each agent, use a unique agent name for each agent as Subject Name (agent name will be overriden if `CONFIG_AGENT_NAME` is not set). Following is simplied agent self-signed certificate generation command tied to it's name;

Author: ```suggestion
Similarly, if you need to generate agent specific keys for each agent, use a unique agent name for each agent as Subject Name (agent name will be overridden if `CONFIG_AGENT_NAME` is not set). 

Following is a simplified command to generate a self-signed certificate tied to an agent's name: 
```

Author: @ravikiran-sulikeri - this was your sentence originally - does this capture the intended meaning?

Reviewer 2: yes

> Similarly, if you need to generate agent specific keys for each agent, use a unique agent name for each agent as Subject Name (agent name will be overriden if `CONFIG_AGENT_NAME` is not set). Following is simplied agent self-signed certificate generation command tied to it's name;

Reviewer: ```suggestion
Similarly, if you need to generate agent-specific keys for each agent, use a unique agent name for each agent as Subject Name (agent name will be overridden if `CONFIG_AGENT_NAME` is not set). 
Following is a simplified agent self-signed certificate generation command tied to its name:
```
Please check the last sentence, I'm not sure if I fixed it in the way it was intended.
